test/agent_otel_merging/agent_otel_merging_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package agent_otel_merging
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/util/common"
+)
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags()
+}
+
+// Tests merging of agent and OTEL yaml
+// Merges OTLP receiver to Agent yaml
+// And Checks if test_metric was generated(YAML merged successfully)
+func TestOtelMerging(t *testing.T) {
+	startAgent(t)
+	appendOtelConfig(t)
+	sendPayload(t)
+	verifyMetricsInCloudWatch(t)
+	verifyHealthCheck(t)
+}
+
+func startAgent(t *testing.T) {
+	common.CopyFile(filepath.Join("agent_configs", "config.json"), common.ConfigOutputPath)
+	require.NoError(t, common.StartAgent(common.ConfigOutputPath, true, false))
+	time.Sleep(10 * time.Second) // Wait for the agent to start properly
+}
+
+func appendOtelConfig(t *testing.T) {
+	cmd := exec.Command("sudo", "/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl",
+		"-a", "append-config", "-s", "-m", "ec2", "-c", "file:"+filepath.Join("resources", "otel.yaml"))
+	output, err := cmd.CombinedOutput()
+	require.NoError(t, err, "Failed to append OTEL config: %s", output)
+	time.Sleep(10 * time.Second) // Wait for the agent to apply the new configuration
+}
+
+func sendPayload(t *testing.T) {
+	currentTimestamp := time.Now().UnixNano()
+	payload, err := os.ReadFile(filepath.Join("resources", "metrics.json"))
+	require.NoError(t, err, "Failed to read JSON payload from config.json")
+	payloadStr := string(payload)
+	payloadStr = strings.ReplaceAll(payloadStr, "$CURRENT_TIMESTAMP", fmt.Sprintf("%d", currentTimestamp))
+
+	curlCmd := exec.Command("curl", "-X", "POST", "http://localhost:4318/v1/metrics",
```

Comment:
Go should have native package that makes HTTP call, I don't think you need to run `exec` here.

Reply:
https://pkg.go.dev/net/http

Reply:
done